### PR TITLE
BUGFIX: NodeType inheritance is not working correct

### PR DESCRIPTION
**Pull request by Fiach**
FIXES: #2663

### CODE REVIEW

**Comment by kitsunet** [comment]
If anyone else looks at this, basically we want to apply enrichment to the full configuration to avoid effects where only the merged configuration would be enriched correctly (eg. one config has the property type, another the inspector config) BUT labels need to be set on the "local" configuration to make sure the package keys match and no duplication happens. So basically two pointcuts are used to allow both to happen.

**Comment by bwaidelich** [comment]
I can't check right now, but please make sure that this doesn't break #2785 – it relies on the fact that the full node type configuration is set once the NodeType instance is created

**Comment by albe** [comment]
So, is there anything blocking from merging right now?

**Comment by mficzel** [comment]
Looks good by reading and makes sense but i want to test later today. Especially to exclude interference with #2785 as bastian mentioned.

**Comment by mficzel** [comment]
There is a merge conflict in the NodeType where the sorting of the NodeType postprocessors collides with this change. The solution is that `$this->fullConfiguration` should become `$fullConfiguration`

File: Neos.Neos/Classes/Aspects/NodeTypeConfigurationEnrichmentAspect.php [hunk 1]
```diff
@@ -56,17 +56,34 @@ class NodeTypeConfigurationEnrichmentAspect
...
+        $joinPoint->setMethodArgument('fullConfiguration', $configuration);
+        $joinPoint->getAdviceChain()->proceed($joinPoint);
+    }
+
+    /**
+     * @Flow\Around("method(Neos\ContentRepository\Domain\Model\NodeType->__construct())")
+     * @param JoinPointInterface $joinPoint
+     * @return void
+     */
+    public function enrichNodeTypeLabelsConfiguration(JoinPointInterface $joinPoint)
```
Comment (daniellienert) [inline Neos.Neos/Classes/Aspects/NodeTypeConfigurationEnrichmentAspect.php:81]:
```suggestion
    public function enrichNodeTypeLabelsConfiguration(JoinPointInterface $joinPoint): void
```

File: Neos.ContentRepository/Classes/Domain/Model/NodeType.php [hunk 2]
```diff
@@ -687,6 +689,16 @@ protected function traverseSuperTypes(NodeType $currentNodeType, $constraintNode
         return null;
     }
 
+    /**
+     * @param array $fullConfiguration
+     */
+    protected function setFullConfiguration(array $fullConfiguration)
```
Comment (daniellienert) [inline Neos.ContentRepository/Classes/Domain/Model/NodeType.php:695]:
```suggestion
    protected function setFullConfiguration(array $fullConfiguration): void
```

**Pull request merged**